design/authenticators_api.md

```diff
@@ -0,0 +1,178 @@
+
+## Feature Overview 
+
+### Use Cases
+
+This feature is divided into 2 personas:
+
+#### Conjur operator
+
+**As a** Conjur operator\
+**I'd like to** know which authenticators are configured correctly\
+**So that** I can complete the configuration properly if it's incomplete (or remove
+it if it's not needed)
+
+#### Conjur user
+
+**As a** Conjur user\
+**I'd like to** know which authenticators are available for authentication\
+**So that** I can authenticate with a properly configured one
+
+Note that both personas would like to know the same thing (configured correctly == available for authentication)
+but the action they need it for is different. 
+
+We can learn 2 things from the above:
+
+1. The Conjur user needs to authenticate according to the response. So this request must be applicable **without** a Conjur access token. 
+1. The Conjur operator needs to know which actions are needed for completing the configuration,
+which may consist sensitive data. So this request must be applicable **only with** a Conjur access token.
+
+This leads us to 2 API endpoints:
+
+1. `/authenticators` (for Conjur users)
+2. `/authenticators/health` (for Conjur operators)
+
+### Process Logic
+
+In the following example, the authenticators `authn` & `authn-1` are properly configured
+and `authn-2` is not whitelisted in the ENV.
+
+#### `/authenticators` Endpoint
+
+- A Conjur user runs an `/authenticators` request:\
+`GET /authenticators`
+- The user gets a response with code 200 with the following body:
+```
+{
+   “authenticators”:
+   [
+      "authn",
+      "authn-1"
+   ]
+}
+```
+
+#### `/authenticators/health` Endpoint
+
+- A Conjur operator logs into Conjur (in any authn method) and receives an access token
+- The operator runs an `/authenticators/health` request with the given access token:\
+`GET /authenticators/health`
+- The operator gets a response with code 200 (more info [here](authenticators_health_api.md#response-code-for-unhealthy-authenticators)) with the following body:
+```
+{
+   “authenticators”:
+   {
+      "authn": "ok",
+      "authn-1": "ok",
+      "authn-2": "Authentication::Security::NotWhitelisted: CONJ00004E 'authn-2' is not whitelisted in CONJUR_AUTHENTICATORS", 
+   },
+   "ok": false
+}
+```
+- The operator understands the issue from the response and fixes it
+- The operator runs the request once again until he gets a healthy response
+
+#### `/info` endpoint
+
+Another endpoint we should address is `evoke`'s `/info` endpoint, which contains the data of the `/authenticators` endpoint
+(with other info such as release, version, etc.). This endpoint calls `http://localhost/authenticators` and adds the output to the info
+json. Having the same data in 2 different places is not optimal as it can confuse the user (and developers), so it would make sense to move the
+implementation of the `/authenticators` endpoint to `evoke`, and remove the `/authenticators` endpoint.
```

Comment:
I don't think we should move this endpoint to evoke because health checks will probably be needed/wanted in OSS in the future and also that will leave OSS without this `authenticators` functionality though I'm not sure how feasible this would be on a technical level.

Reply:
that's exactly what i wrote in the following sentence :)